src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -160,26 +163,40 @@ internal static async Task<Stream> GetPythonDeploymentPackage(IEnumerable<string
             }
             else
             {
-                return await InternalPreparePythonDeployment(files, functionAppRoot);
+                return await InternalPreparePythonDeployment(files, functionAppRoot, requirementsTxtFile);
             }
         }
 
-        private static async Task<Stream> InternalPreparePythonDeployment(IEnumerable<string> files, string functionAppRoot)
+        private static async Task<string> FilterRequirementsTxt(string functionAppRoot, string userRequirements)
+        {
+            var userRequirementsContents = FileSystemHelpers.ReadAllLines(userRequirements);
+
+            var tmp = Path.Combine(Path.GetTempPath(), Path.GetTempFileName().Replace(".", string.Empty));
+            FileSystemHelpers.EnsureDirectory(tmp);
+
+            // Filter the packages that are cached irrespective of their versions
+            var newRequirements = userRequirementsContents.Where(req => !Constants.CachedPythonPackages.Any(req.Contains));
```

Comment:
May need to explicitly check instead of doing `contains`. Otherwise, packages named like `azure-functions-wsgi-adapter` may also get ignored.